pkg/secretless/config/v2/config_http_test.go

```diff
@@ -0,0 +1,70 @@
+package v2
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestNewHTTPConfig(t *testing.T) {
+	t.Run("valid config with 'authenticateURLsMatching' string", func(t *testing.T) {
+		configFileContents := []byte(
+			`
+authenticationStrategy: aws
+authenticateURLsMatching: "*"
+`)
+		cfg, _ := newHTTPConfig(configFileContents)
+		assert.Equal(t, cfg.AuthenticateURLsMatching, []string{"*"})
+	})
+
+	t.Run("valid config with 'authenticateURLsMatching' string list", func(t *testing.T) {
+		configFileContents := []byte(
+			`
+authenticationStrategy: aws
+authenticateURLsMatching: 
+ - "*"
+`)
+		cfg, _ := newHTTPConfig(configFileContents)
+		assert.Equal(t, cfg.AuthenticateURLsMatching, []string{"*"})
+	})
+
+	t.Run("config with bad type for 'authenticateURLsMatching'", func(t *testing.T) {
+		configFileContents := []byte(
+			`
+authenticationStrategy: aws
+authenticateURLsMatching: 
+ - true
+ - "meow"
+`)
+		_, err := newHTTPConfig(configFileContents)
+		assert.Error(t, err)
```

Comment:
Not a blocker but probably the error type/message should be asserted too.

Reply:
We should definitely avoid asserting on message string text.  I am personally okay with asserting on a custom error type, though this is a topic of controversy in the community, and Dave Cheney recommends against even this (at least in most cases):

Reply:
Error type should be for sure asserted on even if it's not custom but I agree that error message assertion might be too much of the detail to worry about.